File: VB/EFCore/ExtendedEvents.Module/BusinessObjects/ExtendedEvent.vb
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl.EF

Namespace ExtendedEvents.Module.BusinessObjects

    <DefaultClassOptions>
    Public Class ExtendedEvent
        Inherits [Event]

        Public Overridable Property CustomSimpleTypeField As String

        Public Overridable Property CustomReferenceTypeField As CustomReferenceTypeField
    End Class

    Public Class CustomReferenceTypeField
        Inherits BaseObject

        Public Overridable Property Name As String
    End Class
End Namespace

File: CS/EFCore/ExtendedEvents.Module/BusinessObjects/ExtendedEvent.cs
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl.EF;
using DevExpress.Persistent.Validation;
using DevExpress.Xpo;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExtendedEvents.Module.BusinessObjects; 
[DefaultClassOptions]
public class ExtendedEvent : Event {
    public virtual string CustomSimpleTypeField { get; set; }
    public virtual CustomReferenceTypeField CustomReferenceTypeField { get; set; }
}
public class CustomReferenceTypeField : BaseObject {
    public virtual string Name { get; set; }
}
